backend/dataall/modules/redshift_datasets/services/redshift_connection_service.py

```diff
@@ -57,6 +57,11 @@ def create_redshift_connection(uri, admin_group, data: dict) -> RedshiftConnecti
             RedshiftConnectionService._check_redshift_connection(
                 account_id=environment.AwsAccountId, region=environment.region, connection=connection
             )
+            if connection.redshiftType == RedshiftType.Cluster.value:
```

Comment:
nit: for provisioned redshift cluster connections we call boto3 api `describe_cluster()` 3 times:
- _check_redshift_connection
- get_cluster_namespaceId
- _get_redshift_encryption

Maybe can optimize - but also the service methods support both serverless and cluster connections so may be a bit less straightforward